packages/core/src/server.ts

```diff
@@ -190,7 +190,15 @@ export const createServer = (
         }
 
         res.status(result.status)
-        res.json(result.body)
+
+        // Handle different body types
+        if (Buffer.isBuffer(result.body)) {
+          res.send(result.body)
+        } else if (typeof result.body === 'string') {
+          res.send(result.body)
+        } else {
+          res.json(result.body)
+        }
```

Comment:
We can simplify the logic a little bit here
```suggestion
        if (Buffer.isBuffer(result.body) || typeof result.body === 'string') {
          res.send(result.body)
        } else {
          res.json(result.body)
        }
```